### PR TITLE
Disable toolbar buttons when the opened solution is not a CPP one

### PR DESCRIPTION
**Pull request by RodicaMihaelaVasilescu**

### CODE REVIEW

File: ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs [hunk 1]
```diff
@@ -266,7 +274,22 @@ public int OnAfterCloseSolution(object aPUnkReserved)
 
 
     #region Private Methods
-
+    private bool DoesSolutionContainCppProject()
+    {
+      DTE2 dte2 = Package.GetGlobalService(typeof(DTE)) as DTE2;
```
Comment (Enache-Ionut) [inline ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs]: Please use the `VsServiceProvider` class to get the services you need.

File: ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs [hunk 2]
```diff
@@ -261,12 +266,33 @@ public int OnAfterCloseSolution(object aPUnkReserved)
...
+
+      if (solution == null)
+      {
+        return true;
+      }
+
+      foreach (var project in solution)
+      {
+        var proj = (Project) project;
+        if (proj.FullName.EndsWith(ScriptConstants.kVcxprojExtension))
```
Comment (Enache-Ionut) [inline ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs]: For project type detection is recommended to check the project GUID.

File: ClangPowerTools/ClangPowerTools/Commands/CommandController.cs [hunk 3]
```diff
@@ -392,6 +393,15 @@ public void OnLicenseChanged(object sender, LicenseEventArgs e)
       activeLicense = e.IsLicenseActive;
     }
 
+    public void VisibilityOnBeforeCommand(object sender, EventArgs e)
+    {
+      if (!(sender is OleMenuCommand command))
```
Comment (Enache-Ionut) [inline ClangPowerTools/ClangPowerTools/Commands/CommandController.cs]: Use standard boolean comparison for `if` statements

File: ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs [hunk 4]
```diff
@@ -261,12 +266,33 @@ public int OnAfterCloseSolution(object aPUnkReserved)
       return VSConstants.S_OK;
     }
 
-
     #endregion
 
 
     #region Private Methods
 
+    private bool SolutionDoesNotContainCppProject()
```
Comment (Enache-Ionut) [inline ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs]: I think it's better to use another class for this solution check (Separation of concerns principle & S from SOLID).

File: ClangPowerTools/ClangPowerTools/Commands/CommandController.cs [hunk 5]
```diff
@@ -402,6 +412,13 @@ public void OnBeforeClangCommand(object sender, EventArgs e)
       if (!(sender is OleMenuCommand command))
         return;
 
+      if (areCommandsDisabled)
+      {
+        command.Enabled = false;
+        return;
+      }
+      command.Enabled = true;
```
Comment (Enache-Ionut) [inline ClangPowerTools/ClangPowerTools/Commands/CommandController.cs]: Considering the below checks it's necessary to activate the command at this point?

Comment (RodicaMihaelaVasilescu) [reply ClangPowerTools/ClangPowerTools/Commands/CommandController.cs]: I just wanted to make sure that commands are enabled when they should be. Are they enabled by default when you open a solution?

File: ClangPowerTools/ClangPowerTools/Commands/CommandController.cs (hunk 3 above)
Comment (RodicaMihaelaVasilescu) [reply ClangPowerTools/ClangPowerTools/Commands/CommandController.cs]: Ok, I will do so.

File: ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs (hunk 1 above)
Comment (RodicaMihaelaVasilescu) [reply ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs]: Ok, done.

File: ClangPowerTools/ClangPowerTools/Commands/CommandController.cs (hunk 5 above)
Comment (Enache-Ionut) [reply ClangPowerTools/ClangPowerTools/Commands/CommandController.cs]: Yes. Check the `OnBeforeClangCommand()` method from `CommandController` class.

File: ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs (hunk 2 above)
Comment (RodicaMihaelaVasilescu) [reply ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs]: You're right, I'll try to find a way to check the project GUID.

File: ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs (hunk 4 above)
Comment (RodicaMihaelaVasilescu) [reply ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs]: I added a SolutionManager class with a method that checks if there are cpp projects in solution.

File: ClangPowerTools/ClangPowerTools/Helpers/SolutionInfo.cs [hunk 6]
```diff
@@ -52,6 +54,36 @@ public static bool IsOpenFolderModeActive()
...
+      {
+        if (IsCppProject((Project)project))
+        {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    public static bool IsCppProject(Project project)
```
Comment (Enache-Ionut) [inline ClangPowerTools/ClangPowerTools/Helpers/SolutionInfo.cs]: This method is not related to a `VS Solution`. Let's keep the `SolutionInfo` clean. Maybe a `ProjectInfo` class will be a good idea.

File: ClangPowerTools/ClangPowerTools/Script/ScriptConstants.cs [hunk 7]
```diff
@@ -6,6 +6,8 @@ public class ScriptConstants
   {
     #region Constants
 
+    public const string kCppProjectGuid = "{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}";
```
Comment (Enache-Ionut) [inline ClangPowerTools/ClangPowerTools/Script/ScriptConstants.cs]: Take into consideration the above comment. This is the `GUID` representation for a `C++ Project` in VS.
